qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp

```diff
@@ -92,44 +91,46 @@ void IndirectPlotOptionsView::setupView() {
   m_plotOptions->leIndices->setCompleter(m_completer.get());
 }
 
-void IndirectPlotOptionsView::emitSelectedWorkspaceChanged(QString const &workspaceName) {
-  emit selectedWorkspaceChanged(workspaceName.toStdString());
+void IndirectPlotOptionsView::subscribePresenter(IIndirectPlotOptionsPresenter *presenter) { m_presenter = presenter; }
+
+void IndirectPlotOptionsView::notifySelectedWorkspaceChanged(QString const &workspaceName) {
+  m_presenter->notifyWorkspaceChanged(workspaceName.toStdString());
```

Comment:
Calling the methods of the subscriber without checking their existence reveals a coupling between the two components. 
But in this particular case hope we can assume there will always be a subscriber for the view in the production scenarios. And also since we don't even have the unit tests for the view class this would not cause any issues. If there is a coverity analysis it would complain on this.

Reply:
I think this should be ok. The `m_presenter` held by the view is an interface class (abstract class https://www.tutorialspoint.com/cplusplus/cpp_interfaces.htm ). To pass in a valid presenter, the presenters class must inherit from this interface and implement each of the pure virtual functions. If it doesn't implement each of these pure virtual method, the class will be considered abstract and the compiler will give a warning saying its not possible to instantiate an abstract class.

Due to the `m_presenter` type being an abstract class, it also means that the View isn't coupled to any particular presenter implementation. This lose-coupling to the Presenter allows you to reuse the same View for different presenter implementations, as long as they adhere to the same interface. I think this lose-coupling is OK because we should nearly always expect a one-to-one relationship between a View and a Presenter in the MVP pattern.